### PR TITLE
Pr 137 review

### PR DESCRIPTION
**Pull request by 222448082Ashen**
## Description
This PR contains two changes:
- Added unit tests for bitmap filename retrieval in unit_test_bitmap.cpp (covers `bitmap_filename` behavior).
- Added peer-review guidance document at PEER_REVIEW.md.

## Type of change
- [x] Tests (new unit tests)
- [x] Documentation (new peer review guidelines)
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How has this been tested?
- Ran full CMake/cTest locally:
  - Commands used:
    - mkdir -p projects/cmake/build && cd projects/cmake/build
    - cmake -S .. -B . -DCMAKE_BUILD_TYPE=Debug
    - cmake --build . --config Debug --target skunit_tests
    - ctest -C Debug --output-on-failure -V
  - Result: 80 tests executed, 0 failures (tests passed 100% locally)

## Checklist (Author)
- [x] Tests added (unit tests in unit_test_bitmap.cpp)
- [x] Documentation added (PEER_REVIEW.md)
- [ ] Self-review (please confirm)
- [ ] Code style / linter check (please confirm)
- [ ] New warnings (please confirm none are introduced)

## Files added / modified
- Added: PEER_REVIEW.md (new reviewer guidance)
- Modified: unit_test_bitmap.cpp (added three sections to test `bitmap_filename`)

---

## Review summary / reviewer prompts
- Does the PR clearly state the _type of change_ and link the related issue?
  - The PR itself does not appear to include a linked issue; please add an issue reference if applicable.
- Does the code follow the project's conventions?
  - Yes — the test follows the Catch2 and existing test layout conventions in unit_tests.
- Is behavior fully covered by tests, including edge cases and failure paths?
  - The new tests cover:
    - Correct filename returned for a loaded bitmap
    - Empty string for null bitmap
    - Empty string for newly created bitmap
  - Additional edge cases to consider:
    - Path-handling across platforms (Windows backslashes vs POSIX slashes). Use a more robust string check (see below).
    - Add tests for `bitmap_name(bitmap)` if not already covered (tests currently only cover filename).
- Backwards compatibility:
  - No API/behavior changes — only tests and docs added.
- New dependencies:
  - None.
- Performance/security:
  - No concerns introduced.

---

## Small issues / suggested fixes (actionable)
1. Unit test robustness (recommended change)
   - In unit_test_bitmap.cpp test that extracts the filename uses `filepath.size() - expected_filename.size()` and `substr`. If `filepath` is unexpectedly shorter than `expected_filename` this could underflow / be unsafe.
   - Suggest replacing with a safer check either:
     - C++20 ends_with: `REQUIRE(filepath.ends_with(expected_filename));`
     - Or use `rfind` and check that

### CODE REVIEW

File: coresdk/src/test/unit_tests/unit_test_bitmap.cpp [hunk 1]
```diff
@@ -128,3 +130,39 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
...
+        std::string expected_filename = "rocket_sprt.png";
+
+        // Load a bitmap
+        bitmap bmp = load_bitmap("rocket", expected_filename);
+        REQUIRE(bmp != nullptr);
+        REQUIRE(bitmap_valid(bmp));
+        
+        // Extract filename
+        std::string filepath = bitmap_filename(bmp);
+        size_t idx = filepath.size() - expected_filename.size();
```
Comment (ralphweng2023) [inline coresdk/src/test/unit_tests/unit_test_bitmap.cpp:147]: ###  Unsafe substring extraction

This code has a potential unsigned integer underflow if `filepath` is shorter than `expected_filename`:

```cpp
size_t idx = filepath.size() - expected_filename.size();  // Can underflow!
std::string filename = filepath.substr(idx, expected_filename.size());
```

I would suggest do it this way:
```
std::string filepath = bitmap_filename(bmp);
REQUIRE(filepath.size() >= expected_filename.size());
std::string filename = filepath.substr(filepath.size() - expected_filename.size());
REQUIRE(filename == expected_filename);
```

File: .github/PEER_REVIEW.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+
+## Description
```
Comment (ralphweng2023) [inline .github/PEER_REVIEW.md:2]: Do we need to include this file in the PR? Looks like this has been added by mistake.

**Review by ralphweng2023 — CHANGES_REQUESTED** [review]
## Peer Review Summary

Thanks for this contribution! I've reviewed the PR against the SplashKit peer review checklist. Overall, the unit tests are well-structured and follow the existing conventions, but there are a couple of issues that need to be addressed before approval.

---

### ✅ **Code Quality**

- [x] **Repository**: Correct repository (splashkit-core)
- [x] **Readability**: Code is clear and follows Catch2 conventions
- [x] **Maintainability**: Tests are modular and follow existing test structure

### ✅ **Functionality**

- [x] **Correctness**: Tests cover the main behaviors of `bitmap_filename`
- [x] **Impact on Existing Functionality**: No breaking changes; only adds tests

### ✅ **Testing**

- [x] **Test Coverage**: Basic scenarios covered (loaded bitmap, null bitmap, created bitmap)
- [x] **Test Results**: Author reports 80 tests, 0 failures (100% pass)

### ⚠️ **Issues Found**

#### 1. **Safety Issue** (inline comment added)
The substring extraction in the first test case has a potential unsigned integer underflow. Please see my inline comment with suggested fix.

#### 2. **`.github/PEER_REVIEW.md` File** (inline comment added)
This file appears to contain PR-specific review content rather than general reusable guidelines. Please clarify if this was added by mistake or if it needs to be restructured.


---

### **Decision: Request Changes** ⚠️

Please address the two inline comments before this can be approved. Once fixed, I'll be happy to approve!

Let me know if you have any questions about the suggested changes.
